File: LeetCode/DP/JS/Leetcode646-maximum-length-of-pair-chain.js
var findLongestChain = function(pairs) {
    pairs.sort((a,b) => a[0] - b[0]);
    const n = pairs.length;
    let f = new Array(n).fill(1);
    let ans = 1;
    for(let i = 1;i < n;i++) {
        for(let j = 0;j < i;j++) {
            if(pairs[i][0] > pairs[j][1]) f[i] = Math.max(f[i],f[j] + 1);
        }
        ans = Math.max(ans,f[i]);
    }
    return ans;
};

File: LeetCode/DP/Rust/Leetcode646-maximum-length-of-pair-chain.rs
impl Solution {
  pub fn find_longest_chain(mut pairs: Vec<Vec<i32>>) -> i32 {
    pairs.sort();
    let n = pairs.len();
    let (mut f, mut ans) = (vec![1; n], 1);
    for i in 1..n {
      for j in 0..i {
        if pairs[i][0] > pairs[j][1] {
          f[i] = f[i].max(f[j] + 1);
        }
      }
      ans = ans.max(f[i]);
    }
    ans
  }
}
